src/messages.cpp

```diff
@@ -239,6 +243,112 @@ class messages_impl
             }
         }
 
+        void modify_msg_with_exclamations( std::string &msg, game_message_type const type ) {
+            // add smileys to messages if the player is high on weed
+            Character *p_char = &get_player_character();
+            if( p_char->has_effect( effect_weed_high ) ) {
```

Comment:
Code style nag: you can drop a level of indentation (and make the code more readable) if you negate this condition and turn it into an early return.

Feel free to ignore.